docs/opensap/index.md

```diff
@@ -63,17 +61,193 @@ To inspect the source code of a particular state, either checkout the correspond
+To make following the course more convenient, also if you have no S/4HANA system at hand, you can also use a mock server.
+Please follow the [guide](https://sap.github.io/cloud-s4-sdk-book/pages/mock-odata.html) as supplementary material on how to setup the mock server locally or on SAP Cloud Platform.
+The guide also explains how to [connect your application](https://sap.github.io/cloud-s4-sdk-book/pages/mock-odata.html#connect-your-application-to-the-mock-server) to the mock server.
+
+The final result of this unit can be found in the branch `course/2_2_end`. You can also download the [zip file](https://github.com/SAP/cloud-s4-sdk-book/archive/course/2_2_end.zip).
+
+#### Unit 3: Securing the Extension App
+
+In this unit, there already is a prepared demo application you should work with.
+It is located in the branch `course/2_3_security` of the example repository.
+You can also download the corresponding [zip file](https://github.com/SAP/cloud-s4-sdk-book/archive/course/2_3_security.zip)
+
+Please follow the instructions in the [security setup guide](http://tiny.cc/s4sdk-security-setup) to prepare your account to setup security for the application.
+Please follow this guide before watching the demo starting after slide 7 of this unit.
+
+#### Unit 4: Introducing Multitenancy
+
+As in the last unit, there is already a prepared demo application you should work with.
+It is located in the branch `course/2_4_multitenancy` of example repository.
+You can also download the corresponding [zip file](https://github.com/SAP/cloud-s4-sdk-book/archive/course/2_4_multitenancy.zip)
+
+The relevant commands and steps to follow and try out the demo in the openSAP video can be found [here](https://github.com/SAP/cloud-s4-sdk-book/blob/course/2_4_multitenancy/docs/multitenancy/setup.md).
+
+#### Unit 5: Adding Automated Tests
+
+To follow the demo shown in the video, please start with the version in the git branch `course/2_5_testing`. 
+You can also download the corresponding [zip file](https://github.com/SAP/cloud-s4-sdk-book/archive/course/2_5_testing.zip).
+
+*Note for IntelliJ users Please disable the `argLine` option in File>Settings as shown below.*
+
+![Disable argline](../assets/argline.png)
+
+The final result of this unit can be found in the branch `course/2_5_testing_practice`. 
+You can download the final result as [zip file](https://github.com/SAP/cloud-s4-sdk-book/archive/course/2_5_testing_practice.zip).
+This version is also the starting point if you also want to implement additional tests as suggested in the unit.
+
+### Week 3 Delivering High Quality Extensions
+
+#### Unit 1: Introducing DevOps and Continuous Delivery
+
+Just enjoy watching the video. There is no preparation required.
+
+#### Unit 2: Rapid Setup of Continuous Delivery
+
+Before starting with this unit, please make sure you installed Docker. 
+To do so, follow the [install instructions](https://docs.docker.com/install/) for your operation system.
```

Comment:
Here's a wild idea: maybe tell people that there is an experimental serverless alternative and link to https://blogs.sap.com/2019/02/11/deep-dive-experimental-serverless-cicd-with-sap-s4hana-cloud-sdk-continuous-delivery-toolkit/ ?

Might be an alternative for learners with weak hardware/no access to a hosted linux server

Reply:
IMO it is too early for that. Let's wait until it is a feature of our offering.